app/controllers/insights_cloud/api/machine_telemetries_controller.rb

```diff
@@ -0,0 +1,67 @@
+module InsightsCloud::Api
+  class MachineTelemetriesController < ::Api::V2::BaseController
+    layout false
+
+    include ::InsightsCloud::ClientAuthentication
+    include ::InsightsCloud::CandlepinCache
+
+    before_action :ensure_telemetry_enabled_for_consumer, :only => [:forward_request]
+    before_action :cert_uuid, :ensure_org, :ensure_branch_id, :only => [:forward_request, :branch_info]
+
+    skip_after_action :log_response_body, :only => [:forward_request]
+    skip_before_action :check_media_type, :only => [:forward_request]
+
+    # The method that "proxies" requests over to Cloud
+    def forward_request
+      certs = candlepin_id_cert @organization
+      cloud_response = ::ForemanRhCloud::CloudRequestForwarder.new.forward_request(request, controller_name, @branch_id, certs)
+
+      if cloud_response.code == 401
+        return render json: {
+          :message => 'Authentication to the Insights Service failed.',
+          :headers => {},
+        }, status: :bad_gateway
+      end
+
+      if cloud_response.headers[:content_disposition]
+        return send_data cloud_response, disposition: cloud_response.headers[:content_disposition], type: cloud_response.headers[:content_type]
+      end
+
+      assign_header(response, cloud_response, :x_resource_count, true)
+      assign_header(response, cloud_response, :x_rh_insights_request_id, false)
+
+      render json: cloud_response, status: cloud_response.code
+    end
+
+    def branch_info
+      render :json => ForemanRhCloud::BranchInfo.new.generate(@uuid, @host, @branch_id, request.host).to_json
+    end
+
+    def assign_header(res, cloud_res, header, transform)
+      header_content = cloud_res.headers[header]
+      return unless header_content
+      new_header = transform ? header.to_s.tr('_', '-') : header.to_s
+      res.headers[new_header] = header_content
+    end
+
+    def ensure_telemetry_enabled_for_consumer
+      render_message 'Telemetry is not enabled for your organization', :status => 403 if Setting[:rh_cloud_disable_insights_proxy]
+    end
+
+    private
+
+    def cert_uuid
+      @uuid ||= User.current.login
+    end
+
+    def ensure_org
+      @organization = @host.organization
```

Comment:
This line fails for `redhat-access` plugin, when trying to configure the service and press "Check connection" button. The log is:

```
16:28:19 rails.1   | 2020-11-16T16:28:19 [I|app|215ee1cc] Started GET "/redhat_access/r/insights/view/api/v3/me" for 127.0.0.1 at 2020-11-16 16:28:19 +0000
16:28:19 rails.1   | 2020-11-16T16:28:19 [I|app|215ee1cc] Processing by InsightsCloud::Api::MachineTelemetriesController#forward_request as */*
16:28:19 rails.1   | 2020-11-16T16:28:19 [I|app|215ee1cc]   Parameters: {"path"=>"view/api/v3/me"}
16:28:19 rails.1   | 2020-11-16T16:28:19 [W|app|215ee1cc] Action failed
16:28:19 rails.1   | 2020-11-16T16:28:19 [I|app|215ee1cc] Backtrace for 'Action failed' error (NoMethodError): undefined method `organization' for nil:NilClass
16:28:19 rails.1   |  215ee1cc | /home/vagrant/foreman_rh_cloud/app/controllers/insights_cloud/api/machine_telemetries_controller.rb:58:in `ensure_org'
```

The same stack is produced by Insights -> Systems, if you have the username configured.